app/Http/Controllers/License/LicenseController.php

```diff
@@ -43,6 +43,31 @@ private function postCurl($post_url, $post_info)
         return $result;
     }
 
+    private function getCurl($get_url)
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $get_url);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 90);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        if (curl_exec($ch) === false) {
+            echo 'Curl error: '.curl_error($ch);
+        }
+        $content = curl_exec($ch);
+        curl_close($ch);
+
+        return json_decode($content, true);
+    }
+
+    public function getLicensekey()
+    {
+        $url = $this->url;
+        $api_key_secret = $this->api_key_secret;
+        $token = $this->token;
+        $getkey = $this->getCurl($url.'api/admin/viewApiKeys?token='.$token);
```

Comment:
you can use $this->token directly, only create a variable if it's used in multiple places